src/main/java/com/iexec/core/task/update/TaskUpdateRequestManager.java

```diff
@@ -102,12 +104,9 @@ public TaskUpdateRequestManager(TaskService taskService) {
         return CompletableFuture.supplyAsync(publishRequest, executorService);
     }
 
-    /**
-     * Authorize one TaskUpdateRequest consumer subscription at a time.
-     * @param consumer
-     * @return
-     */
-    public void setRequestConsumer(final TaskUpdateRequestConsumer consumer) {
-        this.consumer = consumer;
+    private void updateTask(String chainTaskId) {
```

Comment:
Maybe add a todo note for using `ContextuaLockRunner`

Reply:
Good idea! See https://github.com/iExecBlockchainComputing/iexec-core/pull/484/commits/4b0847fb0d03a45df49dd33d66961a3550ea210c.